packages/ckeditor5-table/tests/tableproperties/tablepropertiesediting.js

```diff
@@ -228,6 +228,38 @@ describe( 'table properties', () => {
 					assertTRBLAttribute( table, 'borderStyle', 'solid' );
 					assertTRBLAttribute( table, 'borderWidth', '1px' );
 				} );
+
+				// https://github.com/ckeditor/ckeditor5/issues/8393.
+				it( 'should not upcast contents of nested table', () => {
```

Comment:
This description is too vague. What you're checking here is not whether the nested table is being upcasted. But whether the converter applies exactly to the element on which the border is set.

However, to avoid having to explain this, I'd simply merge this test with the above one. They do the same thing, but check it differently. The former one is just too leaky – it checks something but not enough to catch the issue that we had. So, why not just adding additional check there (the one you added below)?

Also, you should also add a TC for the case from the actual ticket that you're working on. It's a bit different because it features an empty cell.

Finally, perhaps you can think of more cases that should be covered?

Reply:
The other case I can come up with is when we provide the markup with empty nested table, e.g.

```
<table>
<tbody>
<tr>
<td>

<table style="border-bottom: 0 solid #fff;"></table>

</td>
</tr>
</tbody>
</table>
```

This seems like a more general case - an element inside the cell that's not allowed there but has styles. This will trigger the same error as in the issue.